### PR TITLE
Don't use pass by reference to const for cheaply-copied types (bool, char, etc.)

### PR DESCRIPTION
**Pull request by practicalswift**
Don't use pass by reference to const for cheaply-copied types (`bool`, `char`, etc.).

### CODE REVIEW

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -822,7 +822,7 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     /**
      * populate vCoins with vector of available COutputs.
      */
-    void AvailableCoins(std::vector<COutput>& vCoins, bool fOnlySafe=true, const CCoinControl *coinControl = NULL, const CAmount& nMinimumAmount = 1, const CAmount& nMaximumAmount = MAX_MONEY, const CAmount& nMinimumSumAmount = MAX_MONEY, const uint64_t& nMaximumCount = 0, const int& nMinDepth = 0, const int& nMaxDepth = 9999999) const;
+    void AvailableCoins(std::vector<COutput>& vCoins, bool fOnlySafe=true, const CCoinControl *coinControl = NULL, const CAmount& nMinimumAmount = 1, const CAmount& nMaximumAmount = MAX_MONEY, const CAmount& nMinimumSumAmount = MAX_MONEY, const uint64_t& nMaximumCount = 0, const int nMinDepth = 0, const int nMaxDepth = 9999999) const;
```
Comment (TheBlueMatt) [inline src/wallet/wallet.h]: Might as well do it for nMaximumCount as well - copying a 64-bit integer is gonna be faster than pass-by-reference, probably.

Comment (practicalswift) [inline src/wallet/wallet.h]: Fixed! :-)

**Comment by TheBlueMatt** [comment]
utACK cb0e62fb7338cb592508a9dc1624855ca9c8a769

**Review by promag — COMMENTED** [review]
Was `memusage.h` ignored on purpose? Otherwise ACK.

**Comment by practicalswift** [comment]
Rebased!

**Review by TheBlueMatt — COMMENTED** [review]
utACK

**Comment by theuni** [comment]
utACK 99ba0c30087dc45a3805f0cac6d2e534ec52f039

**Comment by laanwj** [comment]
utACK 99ba0c3

**Pull request merged**